src/errors.rs

```diff
@@ -50,51 +50,28 @@ other_error!(google_cloud_storage::client::google_cloud_auth::error::Error);
 other_error!(aws_sdk_s3::Error);
 #[cfg(feature = "server-aws")]
 other_error!(aws_sdk_s3::primitives::ByteStreamError);
-#[cfg(feature = "server-gcp")]
-other_error!(reqwest::Error);
 
 impl<T: Sync + Send + 'static> From<tokio::sync::mpsc::error::SendError<T>> for Error {
     fn from(err: tokio::sync::mpsc::error::SendError<T>) -> Self {
         Self::Other(err.into())
     }
 }
 
-/// Convert ureq errors more carefully
-#[cfg(feature = "server-sync")]
-impl From<ureq::Error> for Error {
-    fn from(ureq_err: ureq::Error) -> Self {
-        match ureq_err {
-            ureq::Error::Status(status, response) => {
-                let msg = format!(
-                    "{} responded with {} {}",
-                    response.get_url(),
-                    status,
-                    response.status_text()
-                );
-                Self::Server(msg)
-            }
-            ureq::Error::Transport(_) => Self::Server(ureq_err.to_string()),
+/// Convert reqwest errors more carefully
+#[cfg(feature = "http")]
+impl From<reqwest::Error> for Error {
+    fn from(err: reqwest::Error) -> Self {
+        if let Some(status_code) = err.status() {
+            let msg = format!(
+                "{} responded with {} {}",
+                err.url().map(|u| u.as_str()).unwrap_or("unknown"),
+                status_code.as_u16(),
+                status_code.canonical_reason().unwrap_or("unknown"),
+            );
+            return Self::Server(msg);
         }
+        Self::Server(err.to_string())
     }
 }
 
 pub(crate) type Result<T> = std::result::Result<T, Error>;
-
-#[cfg(test)]
-mod test {
-    #[cfg(feature = "server-sync")]
-    use super::*;
-
-    #[cfg(feature = "server-sync")]
-    #[test]
-    fn ureq_error_status() {
-        let err = ureq::Error::Status(
-            418,
-            ureq::Response::new(418, "I Am a Teapot", "uhoh").unwrap(),
-        );
-        assert_eq!(
-            Error::from(err).to_string(),
-            "Server Error: https://example.com/ responded with 418 I Am a Teapot"
-        );
-    }
-}
```

Comment:
Seems like we could be testing the reqwest_error_status here.

Reply:
I couldn't find any straightforward way to create a new reqwest::Error, besides doing something that would involve network access, and that seems like it might be flaky.